webapp/src/scriptmanager.tsx

```diff
@@ -424,11 +435,25 @@ export class ScriptManagerDialog extends data.Component<ScriptManagerDialogProps
 
         const zipName = `makecode-${targetNickname}-project-download.zip`
 
-        pxt.BrowserUtils.browserDownloadDataUri(datauri, zipName);
 
         this.setState({
```

Comment:
Was the reordering of `download: null` state and `browserDownloadDataUri` intentional? My assumption would be that we should stay in a downloading state until all the steps have finished, but I think I may not be understanding what `browserDownloadDataUri` does...

Reply:
You're right, I should  move this back, thanks for pointing that out. I think I was reordering stuff to better understand how everything interacted and moving it didn't change anything. 

The `browserDownloadDataUri` function is doing the actual steps to initiate the download of the content to the user's computer. The way we do that differs per method and browser, but regardless, that function holds the download action. For the in-game, chromium case, we are basically putting the contents in an `<a>` tag and then clicking the link to initiate the download. The logic in `handleDownloadAsync` is basically only handling the zipping part.